### PR TITLE
[ENH] Define "modality" and clarify "data type"

### LINKED ISSUE #584: What to call modality directory vs the modality entity?
### **Problem**
Currently in the standard, there is this notion of a `modality`, which I totally assumed meant `eeg`, `meg`, `ieeg`, `anat`, etc. So therefore they would be the "modality" directories that followed subject and session. For example:

`sub-01/ses-01/<modality>/`

However, there is also a `modality` entity which is primarily used in `anat` sub-directories to differentiate various imaging modalities. This shows up like `sub-01/ses-01/anat/sub-01_ses-01_mod-T2w_defacemask.nii.gz`.

**Anat using `modality` in different ways:**
https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/01-magnetic-resonance-imaging-data.html#anatomy-imaging-data

**Entity table:**
https://bids-specification.readthedocs.io/en/stable/99-appendices/04-entity-table.html

### **Expectation**
It would be helpful to have the specification explicitly define what to call the directory level names vs. the filename specific `modality` entity. Since calling them both `modality` leads to a bit of confusion since there is no good 1-1 matching.

Idk what's the right answer here so would be helpful to get some thoughts and then I can PR if it's feasible :p.

### **Additional Info**
See a problem that surfaced in mne-bids at: https://github.com/mne-tools/mne-bids/issues/519#issuecomment-682028237

cc @jasmainak @sappelhoff @hoechenberger @agramfort

### LINKED ISSUE #586: Define "modality" in Common principles
This stems from #571 and #584. Essentially, there is some confusion about terminology for different elements of the specification that could be reduced by defining "modality" and differentiating it from "data type". Of course, the actual definition of modality is up for debate.

I think it would be best to define modality as a subtype of suffix. The reasons for this are:
- The term `modality_label` is used in the anatomical MRI section as a stand-in for all of the anatomical MRI-specific suffixes.
- The "associated modality" entity (`mod`) refers to the suffix of the file from which a defacing mask was created.
- Suffixes are not, however, all "modalities". See https://github.com/bids-standard/bids-specification/issues/584#issuecomment-682330359 and https://github.com/bids-standard/bids-specification/issues/571#issuecomment-673659040.

### PR DESCRIPTION
**Pull request by effigies**
Closes #584.
Closes #586.

Reworded slightly from initial suggestion.

### CODE REVIEW

File: src/02-common-principles.md
Comment (hoechenberger) [inline src/02-common-principles.md:65]:
```suggestion
    For passive recording techniques, such as EEG, MEG, or iEEG,
    the technique is sufficiently uniform to define the modalities `eeg`,
    `meg`, and `ieeg`, respectively.
```

**Review by hoechenberger — APPROVED** [review]

**Comment by tsalo** [comment]
WDYT of also slightly updating the definition of "data types" to clarify that they're reflect in folder names?

**Comment by effigies** [comment]
That one was actually a bit harder to word, IMO. Let me know what you think.

**Review by tsalo — APPROVED** [review]
I really like the updated "Data type" definition. I think this will really help with the confusion about terminology.

**Review by sappelhoff — APPROVED** [review]

**Review by agramfort — APPROVED** [review]

**Comment by tsalo** [comment]
Unless anyone raises any concerns about it, this PR should be good to merge tomorrow (5 days after second approval).

**Comment by sappelhoff** [comment]
> Unless anyone raises any concerns about it, this PR should be good to merge tomorrow (5 days after second approval).

I agree that it's good to be merged, it's a nice clarification.

**Pull request merged**
